tsl/src/continuous_aggs/create.c

```diff
@@ -1550,83 +1548,6 @@ mattablecolumninfo_addentry(MatTableColumnInfo *out, Node *input, int original_q
 	return var;
 }
 
-/*add internal columns for the materialization table */
```

Comment:
This function is currently used by `cagg_create` and by `cagg_rebuild_view_definition`. So I don't think it should be removed.

Reply:
This is the main purpose of this PR, remove the internal column named `chunk_id` so we'll not need it anymore in the following PR for finals form.

Reply:
Confusion points for me:

I thought that the purpose of `cagg_rebuild_view_definition` is to check for errors in an existing (old) cagg. That is, a cagg that still includes the `chunk_id` in the materialization hypertable. And the check for errors is performed by checking if there is a mismatch between the number of columns of the existing (old) cagg’s materialization table and the materialization table that would be generated by the current code (after [this fix](https://github.com/timescale/timescaledb/pull/3899#event-6447088872)), which also includes the `chunk_id` in the materialization hypertable. 

Currently this check uses `mattablecolumninfo_addinternal`, and I haven’t understood how it can still be performed if this function is removed. Won’t it always detect a mismatch and fail in that case?
 
 Apologies if I'm horribly wrong about this 🥲

Reply:
@konskov I made confusion here... you're correct... working to fix it.